src/blob_tree/mod.rs

```diff
@@ -87,6 +102,47 @@ fn resolve_value_handle(tree: &BlobTree, version: &Version, item: InternalValue)
     }
 }
 
+fn resolve_value_handle_owned(
```

Comment:
At this point it would probably be best to remove resolve_value_handle() and only use this function everywhere, otherwise the function is essentially duplicated without much benefit.

And then the _owned suffix can be removed.